### PR TITLE
Directory listing marker fix

### PR DESCRIPTION
**Pull request by tjcelaya**
See #156 

Test plagiarized from `snaplink_test.go`

### CODE REVIEW

**Comment by sean-** [comment]
@tjcelaya once travis blesses this, poke me and I'll approve.

**Comment by tjcelaya** [comment]
poke @sean- 

(derp, mechanically applied the change requested by `goimports` and forgot to change the usage)

File: storage/directory_test.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+	})
+
+	t.Run("successfulWithMarker", func(t *testing.T) {
+		v := url.Values{}
+		v.Set("marker", dirLastEntry)
+
+		testutils.RegisterResponder(
+			"GET",
+			path.Join("/", accountUrl, dirPath)+"?"+v.Encode(),
```
Comment (sean-) [inline storage/directory_test.go]: Can we use `url.URL` here instead?

```go
u := url.URL{}
u.Path = path.Join("/", accountuUrl, dirPath)
u.RawQuery = v.Encode()

testutils.RegisterResponder("GET", u.String(), listDirSuccess)
```

File: storage/directory_test.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+`
+	dirListingPartial = `{"name":"subdirectory1","type":"directory","mtime":"2018-01-01T00:00:00.000Z"}
+`
+	dirLastEntry = "subdirectory1"
+)
+
+func TestList(t *testing.T) {
+	storageClient := &storage.StorageClient{
+		Client: testutils.NewMockClient(testutils.MockClientInput{
+			AccountName: accountUrl,
```
Comment (sean-) [inline storage/directory_test.go]: Can you do a `go-rename` on `accountUrl` and fix it to `accountURL`?  I don't know why `go lint` didn't pick that up at some point in the past.

**Comment by tjcelaya** [comment]
Incorporated suggestions and asked some questions (please expand the comments to see the questions).

**Pull request merged**